src/libstd/io/lazy.rs

```diff
@@ -1,12 +1,10 @@
 use crate::cell::Cell;
 use crate::ptr;
-use crate::sync::Arc;
+use crate::sync::{Arc, RawMutex};
 use crate::sys_common;
-use crate::sys_common::mutex::Mutex;
 
 pub struct Lazy<T> {
-    // We never call `lock.init()`, so it is UB to attempt to acquire this mutex reentrantly!
-    lock: Mutex,
+    lock: RawMutex,
```

Comment:
IIRC, `get` is `unsafe` just because of the reeentrancy problem. Seems like it could be safe now?

Reply:
If it was previously UB to call `get` from within the `init` closure, it will now cause a deadlock. So I guess that is safe, but it would not be recommended..

So maybe remove `unsafe` and update the documentation slightly. Still keep the warning, but not say it's about safety, rather about not getting locked.

Reply:
I should mention that I intentionally did not try to improve the public API of std in this PR. I so far just tried to swap out the backing lock primitives. Then I thought we could do some public facing improvements to the API in later PRs. But we can do this here and now if we agree that's correct.

Reply:
Fair. This is not public API though.

Reply:
Oh. You are right! Yeah, well I'll run some tests with it not being `unsafe` and push here if it works out.